src/storage/src/hummock/compactor/compaction_utils.rs

```diff
@@ -676,3 +663,23 @@ pub fn calculate_task_parallelism_impl(
     let parallelism = compaction_size.div_ceil(parallel_compact_size);
     worker_num.min(parallelism.min(max_sub_compaction as u64) as usize)
 }
+
+pub fn split_watermark_from_task(
+    compact_task: &CompactTask,
+) -> (
+    BTreeMap<u32, TableWatermarks>,
+    BTreeMap<u32, TableWatermarks>,
+) {
+    let mut pk_watermarks = BTreeMap::default();
+    let mut non_pk_prefix_watermarks = BTreeMap::default();
+
+    for (table_id, watermark) in &compact_task.table_watermarks {
+        if let WatermarkSerdeType::PkPrefix = watermark.watermark_type {
+            pk_watermarks.insert(*table_id, watermark.clone());
+        } else {
+            non_pk_prefix_watermarks.insert(*table_id, watermark.clone());
```

Comment:
Can we avoid the watermark clone here by pre-spliting the table watermark into pk prefix watermark and non-pk prefix watermark in the compact task struct here? https://github.com/risingwavelabs/risingwave/blob/7e8f6cddc2ffd0f0fb0c65e110471e5812c07309/src/storage/hummock_sdk/src/compact_task.rs#L200